iothub/service/src/DigitalTwin/Authentication/PnpConnectionStringCredential.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Azure.Devices.Authentication
     /// <summary>
     /// Allows authentication to the API using a Shared Access Key
     /// </summary>
-    internal class SharedAccessKeyCredentials : IotServiceClientCredentials
+    internal class PnpConnectionStringCredential : IotServiceClientCredentials
```

Comment:
Unfortunately all pnp clients have to derive from a specific class that is auto-generated. So instead of using the credential classes we created for IotHub, we have to create new ones. Hence, the prefix 'pnp' to distinguish them.

Reply:
What is special about auth for PnP clients?

Reply:
Nothing special. We just need to derive from a specific class and override a method to add the token to the header. It is just not the same as the one we pass to other clients. Is is sort of duplicate functionality.

Reply:
Oh, duh. Because it is in the DigitalTwinClient, right?

Why can't clients share these types?

Reply:
Because this is a generated type that is passed to the protocol layer. Our other clients are handwritten. I have do a lot of changes to be able to make the other clients use this type. Plus it will mean DT client will have unnecessary AMQP CBS token implementation. DT is only Http.

Reply:
All this should be covered in code comments.

Reply:
Will add